### PR TITLE
MAR-1065: SMS subscription workflow

### PR DESCRIPTION
**Pull request by mchl0208**
A new library was created and we will begin to move to the new user-centric model. So far only two entities have been created (`User`, `Subscription`) but there is more work queued to fully move to the new user-centric models. 

some other additional support modules were added

### CODE REVIEW

File: lib/one_signal/subscription.ex [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+  use OneSignal.Entity
+  import OneSignal.Request
+
+  defp plural_endpoint(id) do
+    "/subscriptions" <> "/#{get_id!(id)}"
+  end
+
+  # defp plural_endpoint(%{alias_label: alias_label, alias_id: alias_id}) do
+  #   "/users/by/#{alias_label}" <> "/#{get_id!(alias_id)}" <> plural_endpoint()
+  # end
```
Comment (silentsilas) [inline lib/one_signal/subscription.ex:11]: Should be good to delete this?

**Review by silentsilas — APPROVED** [review]
Had a very small nitpick, but this otherwise looks great to me!

File: lib/one_signal/subscription.ex (hunk 1 above)
Comment (mchl0208) [reply lib/one_signal/subscription.ex:11]: I would leave it there since the next ticket will use it. :D

**Review by ElijahBrandyberry — COMMENTED** [review]
LGTM!

**Pull request merged**